---
title: "Analysis for AFRI"
author: "Nick Miller"
date: "March 16, 2015"
output:
  html_document:
    theme: cerulean
---

# Get the data for differentially expressed genes responding to insecticides

Note that "character" columns get read as factors.

```{r read_insecticide_data, warning=F, cache=TRUE}
library(xlsx)
thiamethoxam <- read.xlsx("foldchanges.xlsx", 1, colClasses = c("character", "numeric", "character", "character"))
esfenvalerate <- read.xlsx("foldchanges.xlsx", 2, colClasses = c("character", "numeric", "character", "character"))
hist(2^thiamethoxam$log.2.foldchg, main="Thiamethoxam", xlab="fold change", col=colors()[562])
hist(2^esfenvalerate$log.2.foldchg, main="Esfenvalerate", xlab="fold change", col=colors()[630])
```

# Get the correpondences between genes identified by Bansal et al and us

They used a different reference than we did (another argument for having useable genomes). We can see which of their genes correspond to which of ours using blast.
1. Set up a blast db of Sijun's reference transcriptome.
2. Run sequences of differentially expressed genes from Bansal as queries.
3. Identify hits - not that 1 Bansal seq could legitimately hit >1 of Sijun's seqs, but we do want to filter out poor hits (short, mismatches)

```{r read_blast_results}
blast.results <- read.table("blastout")
names(blast.results) <- c("query.id", "subject.id", "pc.identity", "alignment.length", "mismatches", "gap.openings", "q.start", "q.end", "s.start", "s.end", "e-value", "bit.score")
blast.results <- subset(blast.results, alignment.length >= 100)
blast.results <- subset(blast.results, pc.identity >= 98)
```

The fasta file of genes described in the Bansal paper contains 158 contigs, after subsetting blast.results contains `r length(blast.results[,1])` entries. Of the 158 Bansal contigs `r length(unique(blast.results$query.id))` have at least one entry, which isn't to bad = `r round((length(unique(blast.results$query.id))/158) * 100, 1)` percent.

# Get Rag1 up and downregulated genes from Bansal et al

Read in data from supplementary info file.

```{r read_rag1, cache=TRUE}
rag1.up <- read.xlsx("Bansal_diffreged.xlsx", 1)
rag1.down <- read.xlsx("Bansal_diffreged.xlsx", 2)
```

Now we need to get these converted over to the transcript set we are using. First thing to do is extract only those for which we have a blast-based correspondence.

```{r extract_with_correpondnce, cache=TRUE}
rag1.up <- subset(rag1.up, as.character(Seq..Name) %in% unique(as.character(blast.results$query.id)))
rag1.down <- subset(rag1.down, as.character(Seq..Name) %in% unique(as.character(blast.results$query.id)))
```
We can use the inner_join from dplyr function to map Bansal's contig to ours

```{r convert_rag1, warning=F, message=F}
library(dplyr)

correspondences <- blast.results[,1:2]
names(correspondences) <- c("Seq..Name", "transcript")
transcripts.rag1.up <- as.character((inner_join(correspondences, rag1.up, by="Seq..Name"))$transcript)
transcripts.rag1.down <- as.character((inner_join(correspondences, rag1.down, by="Seq..Name"))$transcript)
```

#Compare expression patterns between esfenvalerate and Rag1

Get the esfenvalerate up/down regulated transcripts

```{r}
transcripts.esfenvalerate.up <- as.character(subset(esfenvalerate, log.2.foldchg > 0)$transcript)
transcripts.esfenvalerate.down <- as.character(subset(esfenvalerate, log.2.foldchg < 0)$transcript)


```

Upregulated in both

```{r}
intersect(transcripts.esfenvalerate.up, transcripts.rag1.up)
```

Downregulated in both

```{r}
intersect(transcripts.esfenvalerate.down, transcripts.rag1.down)

```

Up in esfenvalerate, down in Rag1

```{r}
intersect(transcripts.esfenvalerate.up, transcripts.rag1.down)
```

Down in esfenvalerate, up in Rag1

```{r}
intersect(transcripts.esfenvalerate.down, transcripts.rag1.up)
```

So we have a few transcripts that are upregulate in both, what are they?

```{r}
subset(esfenvalerate, transcript %in% intersect(transcripts.esfenvalerate.up, transcripts.rag1.up))$gene

```

All cytochrome P450s

How many different genes from Bansal's original reference do we have here?

```{r}
unique(as.character(subset(correspondences, transcript %in% intersect(transcripts.esfenvalerate.up, transcripts.rag1.up))$Seq..Name))

```

#Compare expression patterns between thiamethoxam and Rag1

Get thiamethoxam up/down regulate transcripts.

```{r}
transcripts.thiamethoxam.up <- as.character(subset(thiamethoxam, log.2.foldchg > 0)$transcript)
transcripts.thiamethoxam.down <- as.character(subset(thiamethoxam, log.2.foldchg < 0)$transcript)

```

Upregulated in both

```{r}
intersect(transcripts.thiamethoxam.up, transcripts.rag1.up)

```

Note that this is also a shared upregulated transcript between esfenvalerate and Rag1.

Downregulated in both

```{r}
intersect(transcripts.thiamethoxam.down, transcripts.rag1.down)

```

What is this?

```{r}
subset(thiamethoxam, transcript %in% intersect(transcripts.thiamethoxam.down, transcripts.rag1.down))$gene
```

An esterase. Note that overporduction of these often implicated in OP resistance.


Up in thiamethoxam down in Rag1

```{r}
intersect(transcripts.thiamethoxam.up, transcripts.rag1.down)

```

Down in thiamethoxam, up in Rag1

```{r}
intersect(transcripts.thiamethoxam.down, transcripts.rag1.up)

```






